File: src/types/payment.ts
import { SurrealDbId } from "../lib/types"

type PaymentType = {
  payment_type: "spending" | "payment"
  payment_method: "كاش" | "فيزا"
  name: string
  category: "products" | "salaries" | "visits" | "expenses"
  amount: number
  visit_id: string | SurrealDbId
}

export type UpdatePaymentData = PaymentType

export type NewPayment = PaymentType & {
  pending: boolean
}

export type Payment = PaymentType & {
  created_at: string
  id: SurrealDbId
  visitor_name: string
  visitor_phone: string
  payment_items: { id: SurrealDbId; amount: number; name: string }[]
}

export type PaymentItem = {
  id: SurrealDbId
  name: string
  amount: number
  visit: SurrealDbId
  payment: SurrealDbId
}

export type PaymentBaseType = {
  amount: number
  category: "products" | "salaries" | "visits" | "expenses"
  created_at: string
  id: SurrealDbId
  name: string
  payment_type: "spending" | "payment"
  payment_method: "كاش" | "فيزا"
  pending: boolean
  visit: SurrealDbId
}

File: src-tauri/src/structs/payment.rs
use serde::{Deserialize, Serialize};
use surrealdb::sql::{Datetime, Thing};

#[derive(Debug, Serialize, Deserialize)]
pub struct Item {
    id: Thing,
    pub name: String,
    pub amount: f64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Payment {
    pub id: Thing,
    payment_type: String,
    name: String,
    category: String,
    pub amount: f64,
    pub payment_method: String,
    visit_id: Thing,
    visitor_name: String,
    visitor_phone: String,
    created_at: Datetime,
    pub payment_items: Option<Vec<Item>>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct OriginalPayment {
    pub amount: f64,
    pub category: String,
    pub created_at: Datetime,
    pub id: Thing,
    pub name: String,
    pub payment_method: String,
    pub payment_type: String,
    pub pending: bool,
    pub visit: Thing,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct PaymentItem {
    id: Thing,
    pub name: String,
    amount: f64,
    pub visit: Thing,
    payment: Thing,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreatePaymentData {
    pub payment_type: String,
    pub name: String,
    pub category: String,
    pub amount: f64,
    pub payment_method: String,
    pub pending: bool,
    pub visit_id: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct CreatePaymentItem {
    pub name: String,
    pub amount: f64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct NewPaymentData {
    pub amount: f64,
    pub category: String,
    pub created_at: Datetime,
    pub id: Thing,
    pub name: String,
    pub payment_method: String,
    pub payment_type: String,
    pub pending: bool,
    pub visit: Thing,
}
